lib/remote/endpoint.hpp

```diff
@@ -21,13 +25,17 @@ class Zone;
  */
 class Endpoint final : public ObjectImpl<Endpoint>
 {
+	friend JsonRpcConnection;
```

Comment:
Instead of this, why not just add a method for it like `IncrementMessageCounter(const ApiFunction::Ptr& ...)` and call it from within that class just like we already do for `m_Clients`.